sure.py

```diff
@@ -423,6 +423,32 @@ def serverType():
 	elif 'Amazon'in server_type or 'Microsoft' in server_type:
 		return 'on-cloud'
 
+#Persona type: COMPUTE/DATA/COMPUTE_AND_DATA
+def personaType():
+	persona_type = re.findall('"([^"]*)"',str(executeCommand('cat /opt/web-app/etc/persona')))[1]
+	return persona_type
+
+#vManage migration issue checks
+def validatenmsBringup():
+    success = False
+    nms_bringup_file = '/opt/web-app/etc/nms_bringup'
+    try:
+        with open(nms_bringup_file, 'r') as csv_file:
+            total_line_num = len(csv_file.readlines())
+        with open(nms_bringup_file, 'r') as csvfile:
+            csv_reader = csv.reader(csvfile)
+            line_count = 0
+            for row in csv_reader:
+                line_count += 1
+            if line_count < total_line_num:
+                success = False
+            else:
+                success = True
+    except Exception as e:
+        log_file_logger.exception(e)
+
+    return success
+
```

Comment:
what is the intent behind this?
1. it is not handling scenario if file does not exist
2. then reading the file 2 times, and comparing the lines vs valid CSV row entries?

Reply:
1) Exception handled in the try catch block and we log the exception here: log_file_logger.exception(e)
in case of exception, the success flag will be false and we fail the test case.
2) We are looking forward for any illegal character in the file /opt/web-app/etc/nms_bringup

Reply:
The way this code is right now, you are checking `line_count` after you've counted the rows a second time.

What is the intention behind this code? What is it supposed to be doing?